doc/searches.rst

```diff
@@ -132,45 +144,389 @@ Processing input
+* nestingLevel: the nesting level at the with the field is missing
+
+GroupsNestingException
+~~~~~~~~~~~~~~~~~~~~~~
+
+The ``Rollerworks\Component\Search\Exception\GroupsNestingException``
+is thrown when the maximum nesting level is exceeded.
+
+This exception provides the following properties:
+
+* maxNesting: Maximum nesting level
+* groupIdx: index of the nested-group that exceeds the maximum nesting level
+* nestingLevel: the nesting level at the which the group is declared
+
+ValuesOverflowException
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``Rollerworks\Component\Search\Exception\ValuesOverflowException``
+is thrown when the maximum number of values is exceeded.
+
+This exception provides the following properties:
+
+* fieldName: Name of the field which has to many values
+* max: Maximum number of values within a field
+* count: Number of values in the field
+* groupIdx: index of the group at which the field was declared
+* nestingLevel: the nesting level at which the field was declared
+
+.. note::
+
+    Not all processors will give the exact number of values.
+
+    FilterQuery will stop further processing when the maximum amount
+    of values is exceeded. But XML, JSON and Array will return the exact
+    number of values.
+
+GroupsOverflowException
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``Rollerworks\Component\Search\Exception\GroupsOverflowException``
+is thrown when the maximum number of group at nesting level is exceeded.
+
+This exception provides the following properties:
+
+* max: Maximum number of subgroups within a (sub)group
+* count: Number of groups in the (sub)group
+* groupIdx: index of the group that exceeds the maximum count
+* nestingLevel: the nesting level at which the group was declared
+
+.. note::
+
+    Not all processors will give the exact number of values.
+
+    FilterQuery will stop further processing when the maximum amount
+    of values is exceeded. But XML, JSON and Array will return the exact
+    number of values.
+
+UnsupportedValueTypeException
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``Rollerworks\Component\Search\Exception\UnsupportedValueTypeException``
+is thrown when you pass value-type to a field that does not support
+that value-type.
+
+This exception provides the following properties:
+
+* fieldName: Name of the field at which the value was declared
+* valueType: Type of the value that is not accepted, eg: range, comparison or pattern-match
+
+InvalidSearchConditionException
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``Rollerworks\Component\Search\Exception\InvalidSearchConditionException``
+is thrown when the SearchCondition has errors.
+
+Most of these errors are eg. failed transformation or invalid range bounds.
+
+This exception provides access to the invalid SearchCondition using ``getCondition()``.
+The actual search value-errors are stored within the ValuesBag of each field.
+
+The following example shows you can render these errors into a display for the user.
+
+.. code-block:: php
+    :linenos:
```

Comment:
Yep, it indicates that the output must have line number.

Everything at the start of a block between `::` is for special options.